### PR TITLE
vNext - Remove CosmosClient.CosmosClientOptions

### LINKED ISSUE #1244: vNext - Remove CosmosClient.CosmosClientOptions
Remove CosmosClient.CosmosClientOptions from public surface.

### PR DESCRIPTION
**Pull request by ealsur**
## Description

Based on API Review, the pipeline should be immutable after being created and exposing publicly the CosmosClientOptions would allow for changes to be made after.

## Type of change

- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Closing issues

Closes #1244

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- [#1256](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/1256) Removed public accessor to CosmosClient.CosmosClientOptions to avoid mutation after pipeline is created
```
Comment (j82w) [inline changelog.md]: Maybe this should go under Fixed since it's removing something?

Comment (ealsur) [reply changelog.md]: Well, the thing is, its not fixing any bug. Its a breaking change.

Comment (j82w) [reply changelog.md]: How about doing a removed tag instead?

### Removed

Comment (ealsur) [reply changelog.md]: @kirankumarkolli Do you have any preference here?

**Pull request merged**